deployments-check.py

```diff
@@ -72,27 +146,101 @@ def main():
         }
 
         for toml_field, addr_field in mapping.items():
-            toml_addr = toml_section.get(toml_field, '').lower()
-            rs_addr = rs_addresses.get(addr_field, '').lower()
-            docs_addr = docs_addresses.get(addr_field, '').lower()
+            toml_addr = str(toml_net.get(toml_field, '') or '').lower()
+            rs_addr = str(rs_addrs.get(addr_field, '') or '').lower()
+            docs_addr = str(docs_addrs.get(addr_field, '') or '').lower()
+
+            # Presence checks (each missing counts as an error)
+            if not toml_addr:
+                print(f"❌ Missing [deployment.{net_key}] {toml_field} in deployment.toml")
+                errors += 1
+            if not rs_addr:
+                print(f"❌ Missing [{net_key}] {addr_field} in crates/boundless-market/src/deployments.rs")
+                errors += 1
+            if not docs_addr:
+                print(f"❌ Missing [{net_key}] {addr_field} in documentation section '{docs_header}'")
+                errors += 1
 
+            # Mismatch checks (only when both sides present)
             if toml_addr and rs_addr and toml_addr != rs_addr:
-                print(f"❌ Mismatch [{toml_key}] {toml_field} between TOML and RS:")
+                print(f"❌ Mismatch [{net_key}] {toml_field} between TOML and RS:")
                 print(f"  TOML: {toml_addr}")
                 print(f"  RS  : {rs_addr}")
                 errors += 1
 
-            if docs_addr and toml_addr and toml_addr != docs_addr:
-                print(f"❌ Mismatch [{toml_key}] {toml_field} between TOML and documentation:")
+            if toml_addr and docs_addr and toml_addr != docs_addr:
+                print(f"❌ Mismatch [{net_key}] {toml_field} between TOML and documentation:")
                 print(f"  TOML: {toml_addr}")
                 print(f"  DOCS: {docs_addr}")
                 errors += 1
 
+    # ---- ZKC + veZKC (only on Ethereum networks) ----
+    for net_key, docs_header in zkc_networks.items():
+        toml_net = toml_section(toml_data, net_key)
+        rs_addrs = extract_rs_addresses(zkc_rs_content, zkc_rs_network_keys[net_key])
+
+        mapping = {
+            'zkc': 'zkc_address',
+            'vezkc': 'vezkc_address',
+            # TODO: add back once we update the deployment.toml
+            # 'zkc-staking-rewards': 'staking_rewards_address',
```

Comment:
Would be nicer to uncomment and have the respective value present in the `deployment.toml` but I guess it's fine for the moment